enhancements/okd-on-fedora-coreos-in-prow.md

```diff
@@ -78,7 +78,7 @@ This is a proposal to:
 ### Graduation Criteria
 
 - [x] [OKD repository](https://github.com/openshift/okd/) exists for community issue and bug triage, and development tracking.
-- [x] The UBI-based container images for all components in the release payload are shared between OKD and OCP for testing in Prow (with a few exceptions, see below)
+- [ ] Container images for all components in the release payload are built atop Centos Stream base images most closely matching the majority of OCP images (as of Feb 2024, CS9) and do not contain any content from RHEL builds of any packages.
```

Comment:
We should add ideas how to enforce it - periodics which can list RPMs and image metadata to prevent the leaks happening again?

>RHEL builds of any packages

I agree on the idea, but I'd prefer the base idea to still be UBI so that the diff would be minimal and we'd have less OKD-specific issues caused by possible "current RHEL" <-> CentOS Stream disrepancies

Reply:
Your motivations for using UBI (stated offline), to maintain bug-for-bug compatibility with OCP, cannot be met without including RHEL content which cannot be redistributed. While we could use UBI as the base but Centos Stream for everything else I don't understand the value that provides given it'll never be bug-for-bug compatible. I would like to move forward with this PR since UBI is not valuable.

I agree that we should seek to reduce duplicate efforts of maintenance and I have confidence that work is in progress. I'd like to move forward with this PR as is today.

Reply:
>While we could use UBI as the base but Centos Stream for everything else I don't understand the value that provides given it'll never be bug-for-bug compatible

I don't mind CentOS Stream used as base. Perhaps using UBI instead would make devs more comfortable with it, but I suppose we can reach an agreement on this.

However its vital to use the same OCP builder if possible/legal - this would help with developer trust (binaries being built with the same compiler as OCP CI) and avoid a costly double rebuild for OKD binaries

Reply:
This still needs to list who's responsible for generating C9S-based images - I'd prefer for this to happen transparently on ART side if possible

Reply:
We cannot use the same builder image as we want the community to write and submit patches, which require them to be able to build. The OCP Builder images have the same restrictions and we can not make them available.

Reply:
@vrutkovs please take a look at the new changes I pushed

Reply:
>We cannot use the same builder image as we want the community to write and submit patches

Community can use a CentOS9 or UBI builder, it doesn't have to be bit-to-bit identical to the builder we use in Prow. This would allow us to use less resources when OKD is being built